src/transformers/models/open_llama/configuration_open_llama.py

```diff
@@ -123,10 +133,48 @@ def __init__(
         self.attention_dropout_prob = attention_dropout_prob
         self.use_stable_embedding = use_stable_embedding
         self.shared_input_output_embedding = shared_input_output_embedding
+        self.rope_scaling = rope_scaling
+        self._rope_scaling_validation()
+
+        # RoPE scaling validation
+        if self.rope_scaling is not None:
+            if not isinstance(self.rope_scaling, dict) or len(self.rope_scaling) != 2:
+                raise ValueError(
+                    f"`rope_scaling` must be a dictionary with with two fields, `name` and `factor`, got {self.rope_scaling}"
+                )
+            rope_scaling_name = self.rope_scaling.get("name", None)
+            rope_scaling_factor = self.rope_scaling.get("factor", None)
+            if rope_scaling_name is None or rope_scaling_name not in ["linear", "ntk", "dynamic"]:
+                raise ValueError(
+                    f"`rope_scaling`'s name field must be one of ['linear', 'ntk', 'dynamic'], got {rope_scaling_name}"
+                )
+            if rope_scaling_factor is None or not isinstance(rope_scaling_factor, float) or rope_scaling_factor <= 1.0:
+                raise ValueError(f"`rope_scaling`'s factor field must be an float > 1, got {rope_scaling_factor}")
+
         super().__init__(
             pad_token_id=pad_token_id,
             bos_token_id=bos_token_id,
             eos_token_id=eos_token_id,
             tie_word_embeddings=tie_word_embeddings,
             **kwargs,
         )
+
+    # Copied from transformers.models.llama.configuration_llama.LlamaConfig._rope_scaling_validation
+    def _rope_scaling_validation(self):
+        """
+        Validate the `rope_scaling` configuration.
+        """
+        if self.rope_scaling is not None:
+            if not isinstance(self.rope_scaling, dict) or len(self.rope_scaling) != 2:
+                raise ValueError(
+                    "`rope_scaling` must be a dictionary with with two fields, `name` and `factor`, "
+                    f"got {self.rope_scaling}"
+                )
+            rope_scaling_type = self.rope_scaling.get("type", None)
+            rope_scaling_factor = self.rope_scaling.get("factor", None)
+            if rope_scaling_type is None or rope_scaling_type not in ["linear", "ntk", "dynamic"]:
+                raise ValueError(
+                    f"`rope_scaling`'s name field must be one of ['linear', 'ntk', 'dynamic'], got {rope_scaling_type}"
+                )
+            if rope_scaling_factor is None or not isinstance(rope_scaling_factor, float) or rope_scaling_factor <= 1.0:
+                raise ValueError(f"`rope_scaling`'s factor field must be an float > 1, got {rope_scaling_factor}")
```

Comment:
nit: instead you can check for none-ness and do an early return meaning the core logic isn't in an indented block

```suggestion
        if self.rope_scaling is None:
            return 
        
        if not isinstance(self.rope_scaling, dict) or len(self.rope_scaling) != 2:
            raise ValueError(
                "`rope_scaling` must be a dictionary with with two fields, `name` and `factor`, "
                f"got {self.rope_scaling}"
            )
        rope_scaling_type = self.rope_scaling.get("type", None)
        rope_scaling_factor = self.rope_scaling.get("factor", None)
        if rope_scaling_type is None or rope_scaling_type not in ["linear", "ntk", "dynamic"]:
            raise ValueError(
                f"`rope_scaling`'s name field must be one of ['linear', 'ntk', 'dynamic'], got {rope_scaling_type}"
            )
        if rope_scaling_factor is None or not isinstance(rope_scaling_factor, float) or rope_scaling_factor <= 1.0:
            raise ValueError(f"`rope_scaling`'s factor field must be an float > 1, got {rope_scaling_factor}")
```